cpp/CMakeLists.txt

```diff
@@ -218,72 +218,78 @@ function(bbp_enable_static_analysis)
       "${CMAKE_BINARY_DIR}"
       "--executable=${ClangTidy_EXECUTABLE}"
       --files-re
-      "${${PROJECT_NAME}_ClangTidy_FILES_RE}"
-      $<$<BOOL:${PROJECT_NAME}_FORMATTING_NO_SUBMODULES>:--git-modules>
+      "${${PROJECT_PREFIX}_ClangTidy_FILES_RE}"
+      $<$<BOOL:${PROJECT_PREFIX}_FORMATTING_NO_SUBMODULES>:--git-modules>
       --excludes-re
-      ${${PROJECT_NAME}_ClangTidy_EXCLUDES_RE}
+      ${${PROJECT_PREFIX}_ClangTidy_EXCLUDES_RE}
       -p
       ${CMAKE_BINARY_DIR}/compile_commands.json
       --action
       check)
-  add_custom_target(clang-tidy ${clang_tidy_command_prefix} -- ${${PROJECT_NAME}_ClangTidy_OPTIONS})
-  if(${PROJECT_NAME}_TEST_STATIC_ANALYSIS)
+  add_custom_target(clang-tidy ${clang_tidy_command_prefix} -- ${${PROJECT_PREFIX}_ClangTidy_OPTIONS})
+  if(${PROJECT_PREFIX}_TEST_STATIC_ANALYSIS)
     add_test(NAME ClangTidy
              COMMAND ${clang_tidy_command_prefix}
                      --make-unescape-re
-                     -- ${${PROJECT_NAME}_ClangTidy_OPTIONS})
+                     -- ${${PROJECT_PREFIX}_ClangTidy_OPTIONS})
   endif()
-  if(${PROJECT_NAME}_ClangTidy_DEPENDENCIES)
-    add_dependencies(clang-tidy ${${PROJECT_NAME}_ClangTidy_DEPENDENCIES})
+  if(${PROJECT_PREFIX}_ClangTidy_DEPENDENCIES)
+    add_dependencies(clang-tidy ${${PROJECT_PREFIX}_ClangTidy_DEPENDENCIES})
   endif()
 
 endfunction(bbp_enable_static_analysis)
 
-bob_option(${PROJECT_NAME}_FORMATTING "Enable helpers to keep CMake and C++ code properly formatted"
+bob_input(${PROJECT_NAME}_CODING_CONV_PREFIX
```

Comment:
> bob_input(${PROJECT_NAME}_CODING_CONV_PREFIX

Coding conventions are only attached to one CMake project. There is no use-case where you use it twice like below:

```cmake
set(project1_CODING_CONV_PREFIX proj1)
set(project2_CODING_CONV_PREFIX proj2)
add_subdirectory(path/to/coding-conventions)
```

Therefore, I would remove `${PROJECT_NAME}` from the variable name. What about renaming the variable `CPPCC_VAR_PREFIX` (which stands of course for _C++ Coding Conventions Variable Prefix_) ?
It would look like below:
```cmake
set(CPPCC_VAR_PREFIX CORENRN)
add_subdirectory(path/to/coding/conventions)
```

Reply:
Besides I am not sure to understand why this variable needs to be a CACHE one? IMO a simple variable is enough:

```cmake
if(NOT CPPCC_VAR_PREFIX)
  set(CPPCC_VAR_PREFIX ${PROJECT_NAME})
endif(NOT CPPCC_VAR_PREFIX)
```
instead of the `bpb_input(...)`

Reply:
What about `HPCCC` for `hpc-coding-conventions`?

Reply:
They are not HPC, but BlueBrain's.
=> `BBPCC` ?

Reply:
Shouldn't this repository be renamed, then?

Reply:
> Shouldn't this repository be renamed, then?

yes. But this is not in the scope of this PR

Reply:
I like `BBPCC`, `BBP_CC` would maybe be a little easier on the eyes?

Reply:
This is supposed to be used as a prefix, like a ${PROJECT_NAME}.
So I prefer `BBPCC_VAR_PREFIX` than `BBP_CC_VAR_PREFIX`

Reply:
OK, less confusion without the implicit reference to `CC`…

Reply:
@matz-e  @tristan0x : It's ok to have `BBP_` prefix there but if we are using this in external projects like NEURON/STEPS, wouldn't it be better to `not` have BBP in there? i.e. setting some CMake variable like `BBPCC_VAR_PREFIX` in STEPS or NEURON repository seems "weird"(?). Also the case if someone else want to incorporate this into their open source projects.

No strong opinion on this - we can go ahead with this but thought good to mention.

Reply:
@pramodk good point!

Reply:
> @matz-e @tristan0x : It's ok to have `BBP_` prefix there but if we are using this in external projects like NEURON/STEPS, wouldn't it be better to `not` have BBP in there? i.e. setting some CMake variable like `BBPCC_VAR_PREFIX` in STEPS or NEURON repository seems "weird"(?). Also the case if someone else want to incorporate this into their open source projects.
> 
> No strong opinion on this - we can go ahead with this but thought good to mention.

There are NumPY, Boost, Google, your name it coding conventions. Now there are BBP's! On the contrary, IMO it is important that a project properly cite the adopted coding conventions.

Reply:
People adopting these coding conventions will cite us anyway by using our code (cloning the submodule, including the files into their repo, ...). It doesn't have to be in the variable name. Also, the `CC` in the variable name bugs me.. evokes too much `C compiler`. I changed ti to something more neutral. I hope that's ok for everyone.

Reply:
I don't understand why a more neutral prefix is better. Besides, it increases the chance of name conflict.